bigchaindb/log/setup.py

```diff
@@ -0,0 +1,154 @@
+        )
+        server_proc.start()
+
+
+def setup_logging(*, user_log_config=None):
+    setup_pub_logger()
+    setup_sub_logger(user_log_config=user_log_config)
+
+
+def create_subscriber_logging_config(*, user_log_config=None):
+    sub_log_config = deepcopy(SUBSCRIBER_LOGGING_CONFIG)
+
+    if not user_log_config:
+        return sub_log_config
+
+    if 'file' in user_log_config:
+        filename = user_log_config['file']
+        sub_log_config['handlers']['file']['filename'] = filename
+
+    if 'level_console' in user_log_config:
+        # TODO error handling - ImpoperlyConfigured
+        level = user_log_config['level_console'].upper()
+        sub_log_config['handlers']['console']['level'] = level
+
+    if 'level_logfile' in user_log_config:
+        # TODO error handling - ImpoperlyConfigured
+        level = user_log_config['level_logfile'].upper()
+        sub_log_config['handlers']['file']['level'] = level
+
+    if 'fmt_console' in user_log_config:
+        fmt = user_log_config['fmt_console']
+        sub_log_config['formatters']['console']['format'] = fmt
+
+    if 'fmt_logfile' in user_log_config:
+        fmt = user_log_config['fmt_logfile']
+        sub_log_config['formatters']['file']['format'] = fmt
+
+    if 'datefmt_console' in user_log_config:
+        fmt = user_log_config['datefmt_console']
+        sub_log_config['formatters']['console']['datefmt'] = fmt
+
+    if 'datefmt_logfile' in user_log_config:
+        fmt = user_log_config['datefmt_logfile']
+        sub_log_config['formatters']['file']['datefmt'] = fmt
+
+    log_levels = user_log_config.get('granular_levels', {})
+
+    for logger_name, level in log_levels.items():
+        # TODO error handling - ImpoperlyConfigured
+        level = level.upper()
+        try:
+            sub_log_config['loggers'][logger_name]['level'] = level
+        except KeyError:
+            sub_log_config['loggers'][logger_name] = {'level': level}
+
+    return sub_log_config
+
+
+class LogRecordStreamHandler(StreamRequestHandler):
+    """Handler for a streaming logging request.
+
+    This basically logs the record using whatever logging policy is
+    configured locally.
+    """
+
+    def handle(self):
+        """
+        Handle multiple requests - each expected to be a 4-byte length,
+        followed by the LogRecord in pickle format. Logs the record
+        according to whatever policy is configured locally.
+        """
+        while True:
+            chunk = self.connection.recv(4)
+            if len(chunk) < 4:
+                break
+            slen = struct.unpack('>L', chunk)[0]
+            chunk = self.connection.recv(slen)
+            while len(chunk) < slen:
+                chunk = chunk + self.connection.recv(slen - len(chunk))
+            obj = self.un_pickle(chunk)
+            record = logging.makeLogRecord(obj)
+            self.handle_log_record(record)
+
+    def un_pickle(self, data):
+        return pickle.loads(data)
```

Comment:
@krish7919 please see error handling in https://github.com/bigchaindb/bigchaindb/pull/1285/commits/68cd50249e942dfbea3622d08eb4a1a9f3e3e044